pkg/models/observer_buffer.go

```diff
@@ -55,6 +55,34 @@ type ObserverBuffer struct {
 	MaxE2ELatency       time.Duration
 	MinE2ELatency       time.Duration
 	SumE2ELatency       time.Duration
+
+	// Using observer buffer to collect metadata
+	ErrorsMetadata ErrorsMetadata
+}
+
+type ErrorsMetadata struct {
+
+	//TODO: Instead of storing a lists here we could have a map with already aggregated/categorized errors and counts per each.
+
+	// This field contains accumulated errors for messages which will be retried.
+	// TODO: do we need that in metadata reporting?
```

Comment:
It may not be used in the front end, but I think it's useful for us to report it.